OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDatesPageFragment.kt

```diff
@@ -0,0 +1,107 @@
+package org.edx.mobile.view
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import org.edx.mobile.R
+import org.edx.mobile.course.CourseAPI
+import org.edx.mobile.databinding.FragmentCourseDatesPageBinding
+import org.edx.mobile.http.HttpStatus
+import org.edx.mobile.http.notifications.FullScreenErrorNotification
+import org.edx.mobile.interfaces.OnDateBlockListener
+import org.edx.mobile.model.course.CourseDateBlock
+import org.edx.mobile.util.BrowserUtil
+import org.edx.mobile.util.DateUtil
+import org.edx.mobile.util.UiUtil
+import org.edx.mobile.view.adapters.CourseDatesAdapter
+import org.edx.mobile.viewModel.CourseDateViewModel
+import org.edx.mobile.viewModel.ViewModelFactory
+import javax.inject.Inject
+
+
+class CourseDatesPageFragment : OfflineSupportBaseFragment() {
+
+    @Inject
+    private lateinit var courseAPI: CourseAPI
+    private lateinit var errorNotification: FullScreenErrorNotification
+
+    private lateinit var binding: FragmentCourseDatesPageBinding
+    private lateinit var viewModel: CourseDateViewModel
```

Comment:
is `CourseDateViewModel` representing the `Repository` in `MVVM` architecture?

if Yes then better to update the name accordingly.